File: sources/kafe/keywords.cpp
#include "keywords.hpp"

namespace kafe
{

    namespace kw
    {
        const std::vector<std::string> keywords = {
            "struct", "fun", "ret", "end",
            "dyn",
            "for", "while", "if",
            "in",
        };

        const std::vector<std::string> bloc_opening_kw = {
            "struct", "fun", "dyn", "for", "while", "if",
        };

    }  // namespace kw

}  // namespace kafe

File: parser/_keywords.py
# coding : utf-8

KEYWORDS = (
    "fun", "ret", "end",
    "while", "for",
    "if", "elif", "else",
    "dyn",
    "in",
)

SPECIALS = (
    ":",
    "--",
    ".",
    "[", "]",
    "(", ")"
)

LIST_DELIMITERS = ("[", "]")

DATA_TYPES = (
    "int",
    "double",
    "str",
    "bool",
    "list",
    "void"
)

BOOLEANS = ("true", "false")

STRING_DELIMITERS = ("\"", "'")

OPERATORS = (
    "+", "-", "*", "/", "%",
    "++", "--",
    "&", "|", "~", "<<", ">>", "^",
    "&&", "||", "!",
    "<", "<=", ">", ">=", "==", "!="
)

COMMENTS = ("#", )


if __name__ == '__main__':
    raise NotImplementedError("This script isn't meant to be run directly")
